adrs/5541831634/README.md

```diff
@@ -0,0 +1,160 @@
+# React project files and directories names conversion
+
+* **State:** Draft
+* **Created:** 2020-12-01
+* **Tags:** react
+
+## Context
+
+Across the ecosystem, we can find all sorts of files and directory naming
+conventions in React projects.
+
+The variety of conventions causes misalignment and shifts focus away from more
+important matters.
+
+Although there are no silver bullets for this topic, committing to a convention
+will reduce cognitive load by triggering pattern recognition, enabling engineers
+to focus their brainpower on their tasks.
+
+## Resolution
+
+* You **must** follow [NodeJS files and directories names conversion](./../3122196229/README.md)
+* You **must** use `<rootDir>/src/@types` to store TypeScript definition files
+  (`*.d.ts`).
+* You **must** use `<rootDir>/src/app.tsx` to be your application root
+  component.
+* You **must** use `<rootDir>/src/polyfills` to store polyfills.
+
+### NextJS
+
+Some resolutions are related to NextJS projects.
+
+* You **must** use `<rootDir>/src/pages` for [NextJS Pages](https://nextjs.org/docs/basic-features/pages)
+  directory.
+* In NextJS project, you **must** re-export a NextJS app from
+  `<rootDir>/src/app.tsx` in `<rootDir>/src/pages/_app.tsx`
+* In NextJS project, you **must** re-export a NextJS Documents from
+  `<rootDir>/src/document.tsx` in `<rootDir>/src/pages/_document.tsx`
+* In NextJS project, you **must** re-export a NextJS Documents from
+  `<rootDir>/src/error.tsx` in `<rootDir>/src/pages/_error.tsx`
+
+### Fractal pattern directories and files
```

Comment:
Haven't worked on a Nextjs project large enough to warrant something like this yet but I'm generally open to it. 
It would be sweet if there was an automated way to enforce the structure